src/Controls/ImmersiveControls.js

```diff
@@ -0,0 +1,109 @@
+import FirstPersonControls from 'Controls/FirstPersonControls';
+import AnimationPlayer, { AnimatedExpression } from 'Core/AnimationPlayer';
+
+// Expression used to damp camera's moves
+function moveCameraExp(root, progress) {
+    const dampingProgress = 1 - Math.pow((1 - (Math.sin((progress - 0.5) * Math.PI) * 0.5 + 0.5)), 2);
+    root.camera.position.lerpVectors(root.positionFrom, root.positionTo, dampingProgress);
+}
+/**
+ * @classdesc Camera controls adapted to immersive view.
+ * It allow the user to navigate threw {@link OrientedImageLayer} data.
+ *
+ * <ul> Bindings inherited from FirstPersonControls
+ * <li><b> up + down keys : </b> forward/backward </li>
+ * <li><b> left + right keys: </b> strafing movements </li>
+ * <li><b> pageUp + pageDown: </b> vertical movements </li>
+ * <li><b> mouse click+drag: </b> pitch and yaw movements (as looking at a panorama) </li>
+ * </ul>
+ * <ul> Bindings added
+ * <li><b> keys Z : </b> Move camera to the next panoramic </li>
+ * </ul>
+ * @extends FirstPersonControls
+ */
+class ImmersiveControls extends FirstPersonControls {
+    /**
+     * @constructor
+     * @param { View } view - View where this control will be used
+     * @param { Object } options - Configuration of this controls
+     * @param { number } options.animationDuration - Duration of the animation to another panoramic.
+     */
+    constructor(view, options = {}) {
+        super(view, options);
+
+        // store layers
+        this.layers = [];
+        this.currentLayerIndex = 0;
+
+        // manage camera movements
+        this.player = new AnimationPlayer();
+        this.animationMoveCamera = new AnimatedExpression({
+            duration: options.animationDuration || 50,
+            root: this,
+            expression: moveCameraExp,
+            name: 'Move camera',
+        });
+        this.player.addEventListener('animation-frame', this.updateView.bind(this));
+    }
+
+    /**
+     * Adds a layer with datas.
+     *
+     * @param {OrientedImageLayer} layer - Add a layer
+     */
+    addLayer(layer) {
+        this.layers.push(layer);
+        this.currentLayer = layer;
+        this.currentLayerIndex = this.layers.length - 1;
```

Comment:
Can you explain the role of `this.currentLayerIndex`?

Reply:
I've removed Layer from ImmersiveControls to simplify the control